components/script/dom/xmlhttprequest.rs

```diff
@@ -295,6 +295,9 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
 
     // https://xhr.spec.whatwg.org/#the-open()-method
     fn Open(&self, method: ByteString, url: USVString) -> ErrorResult {
+        // Step 1
+        if !self.global().r().as_window().Document().r().is_fully_active() { return Err(Error::InvalidState); }
```

Comment:
That concept is not implemented yet. We also will not be able to interact with the `Document` on the worker thread; we would need to block on a runnable sent to the document's associated script thread.